algoliasearch-core/src/test/java/com/algolia/search/serialization/TestAutomaticFacetFilterDeserializer.java

```diff
@@ -0,0 +1,27 @@
+package com.algolia.search.serialization;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.algolia.search.Defaults;
+import com.algolia.search.models.rules.AutomaticFacetFilter;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.Test;
+
+public class TestAutomaticFacetFilterDeserializer {
+
+  @Test
+  void testDeserializer() throws JsonProcessingException {
+    String json = "{\"facet\":\"brand_name\"}";
+    AutomaticFacetFilter automaticFacetFilter =
+        Defaults.getObjectMapper().readValue(json, AutomaticFacetFilter.class);
+    assertThat(automaticFacetFilter.getFacet()).isEqualTo("brand_name");
+  }
+
+  @Test
+  void testDeserializerLegacy() throws JsonProcessingException {
```

Comment:
There is another possible case `["lastname","firstname"]` which is not covered by the test see:

https://github.com/algolia/algoliasearch-client-csharp/blob/f2c10db5f84a141f8e3ae54b9c30a8262db5441a/src/Algolia.Search.Test/Serializer/SerializerTest.cs#L183

Reply:
Thank you for the review! I've added the missing test 👍